### ISSUE TITLE
I update the Android project for tiny-cnn.

### ISSUE THREAD

**Issue by zhangqianhui** [nyanp/tiny-cnn#256]
the links is https://github.com/zhangqianhui/CnnForAndroid

I update the two parts !

1.support caffe model and tiny-cnn net.
2.provide the jni interface for caffe_converter .
3.provide two samples for cnn.

Next What I'm going to do is 

1.update the tiny-cnn version as i use the old now.
2.provide the interface to extract feature.

Give me some advice.

**Comment by edgarriba** [comment]
@zhangqianhui  Nice work!
> 1.update the tiny-cnn version as i use the old now.

Please, go ahead.

>  2.provide the interface to extract feature.

You mean an interface to extract raw data from a certain layer? Check it out https://github.com/nyanp/tiny-cnn/issues/261

**Comment by bhack** [comment]
@zhangqianhui I really prefer that you can integrate your work here with a PR. Jni on caffe_converter it is not a mid term solution. I think that on android and ios it is more and more important to have a serialization format for Tiny. /cc @wangyida

**Comment by bhack** [comment]
See also http://tools.android.com/tech-docs/external-c-builds

**Comment by bhack** [comment]
For an example you can see https://github.com/googlesamples/android-ndk/tree/master-cmake

**Comment by bhack** [comment]
We need also to unify this issue with https://github.com/nyanp/tiny-cnn/issues/251

**Comment by zhangqianhui** [comment]
ok.

**Comment by zhangqianhui** [comment]
@bhack  Do you mean to use make a project to use NDK compile it ?
and for example 
https://github.com/dmlc/mxnet/tree/master/amalgamation

**Comment by bhack** [comment]
I mean that with a PR an android build and example could be integrated here in the main repository. See https://github.com/googlesamples/android-ndk/tree/master/cmake

**Comment by patrikhuber** [comment]
If you need a simple and great solution for serialisation, I can really recommend [cereal](https://github.com/USCiLab/cereal). It's header-only, simple, and just great to use. I'm not 100% sure it builds on iOS, but it only requires >=gcc-4.7 or >=clang-3.3 so I don't see why it wouldn't.

**Comment by pansk** [comment]
I had a quick glance at cereal. Not bad at all!

**Comment by edgarriba** [comment]
@nyanp 
is prototyping serialization with cereal so probably he could give us some hints.
https://github.com/tiny-dnn/dnn-serialize

**Comment by zhangqianhui** [comment]
@edgarriba https://github.com/theSundayProgrammer/tiny_nn_json
I think it is necessary to load  the weight for net from json file. Just like mxnet .

**Comment by patrikhuber** [comment]
I think there might be two separate issues at hand, but I may be mistaken:

1) Save and load trained models to and from the disk - this would involve just storing all data in the network class to the disk, including e.g. weights (this is the prime example for serialisation)

2) Having a "config" file where we can set parameters, and these will then be loaded to train a model. I think we can actually use a very similar or same infrastructure for that, since most or some of the parameters are already encapsulated in the class. I think it should be possible to use the same serialisation approach here.

Should we open a separate issue for this topic? It's certainly a bit in the wrong place here.

**Comment by edgarriba** [comment]
@patrikhuber go ahead!

**Comment by patrikhuber** [comment]
Opened #276.

**Comment by zhangqianhui** [comment]
@bhack https://github.com/zhangqianhui/tiny-cnn/tree/master/examples/Android

**Comment by bhack** [comment]
I don't think that make sense to have android in an external repository. Why do you don't want to integrate here with a PR?

**Comment by zhangqianhui** [comment]
@bhack Now have some problems .

**Comment by bhack** [comment]
I see. You have still also the paths hardcoded https://github.com/zhangqianhui/tiny-cnn/blob/master/examples/Android/jni/Android.mk

**Comment by zhangqianhui** [comment]
@bhack I will change it right now

**Comment by bhack** [comment]
When you can try to organize a PR.

**Comment by zhangqianhui** [comment]
@bhack I already organize a PR

**Comment by bhack** [comment]
OK I think that we can close this and who is interested can follow the review of https://github.com/tiny-dnn/tiny-dnn/pull/283
